CMakeLists.txt

```diff
@@ -547,6 +547,18 @@ else()
   message(FATAL_ERROR "Unknown processor:" ${CMAKE_SYSTEM_PROCESSOR})
 endif()
 
+# If ARCH is originally set as either x86 or x86_64, perform an additional
+# check to determine whether 32-bit or 64-bit is supported. This happens in some
+# cases such as when building in Docker, where the host-level architecture is 64-bit
+# but the Docker image should result in building for a 32-bit architecture.
+if (${ARCH} STREQUAL "x86" OR ${ARCH} STREQUAL "x86_64")
+  if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+    set(ARCH "x86_64")
+  else()
+    set(ARCH "x86")
+  endif()
+endif()
```

Comment:
Should `${ARCH} STREQUAL "x86" OR ` be removed?
https://github.com/xianyi/OpenBLAS/pull/2058/files

Reply:
I didn't even think of using `MATCHES`. It should help me remove a bunch of this

Reply:
Talked offline. Basically, we may just want to check the 2nd condition based on the issue

```
elseif(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "x86_64")
  set(ARCH "x86_64")
elseif(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "amd64")
  set(ARCH "x86_64")
elseif(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "AMD64")
  # cmake reports AMD64 on Windows, but we might be building for 32-bit.
  if(CMAKE_SIZEOF_VOID_P EQUAL 8)
    set(ARCH "x86_64")
  else()
    set(ARCH "x86")
  endif()
```